api/src/main/java/edu/wpi/first/shuffleboard/api/util/Debouncer.java

```diff
@@ -0,0 +1,61 @@
+package edu.wpi.first.shuffleboard.api.util;
+
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * A simple class for debouncing method calls. Debouncing is used to prevent an expensive method from being called
+ * in rapid succession, only allowing it to run after a certain amount of time has passed without it being called
+ * again.
+ */
+public class Debouncer {
+
+  private ScheduledFuture<?> future = null;
+  private final ScheduledExecutorService executorService;
+
+  private final long debounceDelay;
+  private final TimeUnit unit;
+
+  /**
+   * Creates a new debouncer.
+   *
+   * @param debounceDelay the maximum time delta between calls that should be allowed
+   * @param unit          the unit of {@code debounceDelay}
+   */
+  public Debouncer(long debounceDelay, TimeUnit unit) {
+    this.debounceDelay = debounceDelay;
+    this.unit = unit;
+    executorService = Executors.newSingleThreadScheduledExecutor(ThreadUtils::makeDaemonThread);
+  }
+
+  /**
+   * Debounces a function call. Generally, this should only be used to debounce multiple calls to the same method/lambda
+   * expression.
+   *
+   * @param runnable the functional unit to be run
+   */
+  public void debounce(Runnable runnable) {
+    if (future != null && !future.isDone()) {
+      future.cancel(false);
+    }
```

Comment:
Can you write a unit test that actually demonstrates what you are trying to do here. Its not at all clear the behaviour you are expecting out of this class. I think that "Debouncing" is not the correct term but I can't exactly tell because I don't see any tests that show the expected behaviour.